### DISCUSSION TITLE
The new deployment is trying to delete CMK

### DISCUSSION THREAD

**Discussion by lulei2007** [aws/aws-cdk#22693] [Q&A]
we use CDK to build pipeline for deployment, but sometimes the deployment asked to delete the existing CMK.
The CMK should not be deleted, how may I avoid this?

Shall we expect to create CMK if not exists, but not delete and recreate.

**Comment by pahud** [comment]
I think you need to find out why your CMK was deleted. Is the CMK created by the same CDK pipeline? Did you change any CDK code?

**Reply by lulei2007** [reply]
The CMK was created by the same CDK pipeline, under the same account and region.
Even if the CMK was not created by it, CDK should not delete it, right?

Is there any way to create resource if not exist? deleting is not acceptable for some resources like cmk. any encrypted data will be screwed.
